sdk/communication/communication-administration/src/phoneNumber/lro/purchase/operation.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  PurchaseReservationPollOperationState,
+  PurchaseReservationPollOperation,
+  UpdatePollerOptions
+} from "../../lroModels";
+import { isComplete } from "../utils";
+
+/**
+ * @summary Reaches to the service and queries the status of the operation.
+ * @param {PurchaseReservationPollOperation} this The poll operation
+ * @param {UpdatePollerOptions<PurchaseReservationPollOperationState>} [options={}] Additional options for the poll operation
+ */
+async function update(
+  this: PurchaseReservationPollOperation,
+  options: UpdatePollerOptions<PurchaseReservationPollOperationState> = {}
+): Promise<PurchaseReservationPollOperation> {
+  const state = this.state;
+  const { reservationId, client } = state;
+  const requestOptions = state.options || {};
+
+  if (options.abortSignal) {
+    requestOptions.abortSignal = options.abortSignal;
+  }
+
+  try {
+    if (!state.isStarted) {
+      await client.purchaseReservation(reservationId, requestOptions);
+      state.isStarted = true;
+    }
+
+    if (!state.isCompleted) {
+      const result = await client.getReservation(reservationId, requestOptions);
+      state.isCompleted = isComplete(result, "Success");
```

Comment:
For all LROs we probably need to also complete when the enum is of a negative but complete value, like `Error` or `Expired` etc. There's no point in continuing to poll in those cases, we should complete and set the error.

Reply:
`isComplete` actually throws if the status is `Error` or `Expired`